src/compression_wrapper.c

```diff
@@ -132,11 +131,16 @@ typedef struct {
 } ZstdFile;
 #endif
 
+void bufferToHexString(const unsigned char* buffer, int size, char* hexString, int hexStringSize) {
```

Comment:
Since we are passing the `magic` into this can we match the `buffer` and `magic` types? 
For example to just `const char* buffer`.